### PR TITLE
added fee tracking to thala cdp

### PR DESCRIPTION
**Pull request by LawsonGraham**

### CODE REVIEW

**Comment by llamatester** [comment]
The move-dollar.ts adapter exports: 
        
 
 _______________________________________
    Fees for 5/7/2023
    _______________________________________
    
    APTOS 👇
    Backfill start time: 4/4/2023
    NO METHODOLOGY SPECIFIED
    Total fees: 42161.35659268602
    Daily fees: 0.027692499160766604
    Timestamp: 1688601598
    
    
    
    
    Done in 5.28s.

**Comment by llamatester** [comment]
The move-dollar.ts adapter exports: 
        
 
 _______________________________________
    Fees for 5/7/2023
    _______________________________________
    
    APTOS 👇
    Backfill start time: 4/4/2023
    NO METHODOLOGY SPECIFIED
    Total fees: 42161.35659268602
    Daily fees: 0.027692499160766604
    Timestamp: 1688601598
    
    
    
    
    Done in 4.43s.

File: fees/move-dollar.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import fetchURL from "../utils/fetchURL";
+import { SimpleAdapter } from "../adapters/types";
+import { CHAIN } from "../helpers/chains";
+
+const thalaDappURL = 'https://app.thala.fi/';
+const feesQueryURL = `${thalaDappURL}/api/protocol-fee-chart?timeframe=`;
+
+const feesEndpoint = (startTimestamp: number, timeframe: string) => 
+startTimestamp ? feesQueryURL + timeframe + `&startTimestamp=${startTimestamp}` : feesQueryURL + timeframe;
```
Comment (dtmkeng) [inline fees/move-dollar.ts]: is can query `start` and `end` time, if you set like this it can not backfill data

**Comment by llamatester** [comment]
The thalaswap adapter exports: 
        
 
 _______________________________________
    Dexs for 9/7/2023
    _______________________________________
    
    Failed to get value on aptos: Error: connect ECONNREFUSED ::1:3000
    APTOS 👇
    Timestamp attempted: 8/7/2023
    AxiosError: connect ECONNREFUSED ::1:3000
        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
      port: 3000,
      address: '::1',
      syscall: 'connect',
      code: 'ECONNREFUSED',
      errno: -111,
      config: {
        transitional: {
          silentJSONParsing: true,
          forcedJSONParsing: true,
          clarifyTimeoutError: false
        },
        adapter: [Function: httpAdapter],
        transformRequest: [ [Function: transformRequest] ],
        transformResponse: [ [Function: transformResponse] ],
        timeout: 0,
        xsrfCookieName: 'XSRF-TOKEN',
        xsrfHeaderName: 'X-XSRF-TOKEN',
        maxContentLength: -1,
        maxBodyLength: -1,
        env: { FormData: [Function] },
        validateStatus: [Function: validateStatus],
        headers: {
          Accept: 'application/json, text/plain, */*',
          'User-Agent': 'axios/0.27.2'
        },
        method: 'get',
        url: 'http://localhost:3000/api/trading-volume-chart?timeframe=1D&endTimestamp=1688947198',
        data: undefined
      },
      request: <ref *1> Writable {
        _writableState: WritableState {
          objectMode: false,
          highWaterMark: 16384,
          finalCalled: false,
          needDrain: false,
          ending: false,
          ended: false,
          finished: false,
          destroyed: false,
          decodeStrings: true,
          defaultEncoding: 'utf8',
          length: 0,
          writing: false,
          corked: 0,
          sync: true,
          bufferProcessing: false,
          onwrite: [Function: bound onwrite],
          writecb: null,
          writelen: 0,
          afterWriteTickInfo: null,
          buffered: [],
          bufferedIndex: 0,
          allBuffers: true,
          allNoop: true,
          pendingcb: 0,
          constructed: true,
          prefinished: false,
          errorEmitted: false,
          emitClose: true,
          autoDestroy: true,
          errored: null,
          closed: false,
          closeEmitted: false,
          [Symbol(kOnFinished)]: []
        },
        _events: [Object: null prototype] {
          response: [Function: handleResponse],
          error: [Function: handleRequestError],
          socket: [Function: handleRequestSocket]
        },
        _eventsCount: 3,
        _maxListeners: undefined,
        _options: {
          maxRedirects: 21,
          maxBodyLength: 10485760,
          protocol: 'http:',
          path: '/api/trading-volume-chart?timeframe=1D&endTimestamp=1688947198',
          method: 'GET',
          headers: [Object],
          agent: undefined,
          agents: [Object],
          auth: undefined,
          hostname: 'localhost',
          port: '3000',
          nativeProtocols: [Object],
          pathname: '/api/trading-volume-chart',
          search: '?timeframe=1D&endTimestamp=1688947198'
        },
        _ended: true,
        _ending: true,
        _redirectCount: 0,
        _redirects: [],
        _requestBodyLength: 0,
        _requestBodyBuffers: [],
        _onNativeResponse: [Function (anonymous)],
        _currentRequest: ClientRequest {
          _events: [Object: null prototype],
          _eventsCount: 7,
          _maxListeners: undefined,
          outputData: [],
          outputSize: 0,
          writable: true,
          destroyed: false,
          _last: true,
          chunkedEncoding: false,
          shouldKeepAlive: false,
          maxRequestsOnConnectionReached: false,
          _defaultKeepAlive: true,
          useChunkedEncodingByDefault: false,
          sendDate: false,
          _removedConnection: false,
          _removedContLen: false,
          _removedTE: false,
          strictContentLength: false,
          _contentLength: 0,
          _hasBody: true,
          _trailer: '',
          finished: true,
          _headerSent: true,
          _closed: false,
          socket: [Socket],
          _header: 'GET /api/trading-volume-chart?timeframe=1D&endTimestamp=1688947198 HTTP/1.1\r\n' +
            'Accept: application/json, text/plain, */*\r\n' +
            'User-Agent: axios/0.27.2\r\n' +
            'Host: localhost:3000\r\n' +
            'Connection: close\r\n' +
            '\r\n',
          _keepAliveTimeout: 0,
          _onPendingData: [Function: nop],
          agent: [Agent],
          socketPath: undefined,
          method: 'GET',
          maxHeaderSize: undefined,
          insecureHTTPParser: undefined,
          joinDuplicateHeaders: undefined,
          path: '/api/trading-volume-chart?timeframe=1D&endTimestamp=1688947198',
          _ended: false,
          res: null,
          aborted: false,
          timeoutCb: null,
          upgradeOrConnect: false,
          parser: null,
          maxHeadersCount: null,
          reusedSocket: false,
          host: 'localhost',
          protocol: 'http:',
          _redirectable: [Circular *1],
          [Symbol(kCapture)]: false,
          [Symbol(kBytesWritten)]: 0,
          [Symbol(kNeedDrain)]: false,
          [Symbol(corked)]: 0,
          [Symbol(kOutHeaders)]: [Object: null prototype],
          [Symbol(errored)]: null,
          [Symbol(kUniqueHeaders)]: null
        },
        _currentUrl: 'http://localhost:3000/api/trading-volume-chart?timeframe=1D&endTimestamp=1688947198',
        [Symbol(kCapture)]: false
      }
    }
    
    
    
    
    Done in 11.81s.

**Comment by llamatester** [comment]
The move-dollar.ts adapter exports: 
        
 
 _______________________________________
    Fees for 9/7/2023
    _______________________________________
    
    Failed to get value on aptos: Error: connect ECONNREFUSED ::1:3000
    APTOS 👇
    Timestamp attempted: 8/7/2023
    AxiosError: connect ECONNREFUSED ::1:3000
        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
      port: 3000,
      address: '::1',
      syscall: 'connect',
      code: 'ECONNREFUSED',
      errno: -111,
      config: {
        transitional: {
          silentJSONParsing: true,
          forcedJSONParsing: true,
          clarifyTimeoutError: false
        },
        adapter: [Function: httpAdapter],
        transformRequest: [ [Function: transformRequest] ],
        transformResponse: [ [Function: transformResponse] ],
        timeout: 0,
        xsrfCookieName: 'XSRF-TOKEN',
        xsrfHeaderName: 'X-XSRF-TOKEN',
        maxContentLength: -1,
        maxBodyLength: -1,
        env: { FormData: [Function] },
        validateStatus: [Function: validateStatus],
        headers: {
          Accept: 'application/json, text/plain, */*',
          'User-Agent': 'axios/0.27.2'
        },
        method: 'get',
        url: 'http://localhost:3000/api/protocol-fee-chart?timeframe=1D&endTimestamp=1688947198',
        data: undefined
      },
      request: <ref *1> Writable {
        _writableState: WritableState {
          objectMode: false,
          highWaterMark: 16384,
          finalCalled: false,
          needDrain: false,
          ending: false,
          ended: false,
          finished: false,
          destroyed: false,
          decodeStrings: true,
          defaultEncoding: 'utf8',
          length: 0,
          writing: false,
          corked: 0,
          sync: true,
          bufferProcessing: false,
          onwrite: [Function: bound onwrite],
          writecb: null,
          writelen: 0,
          afterWriteTickInfo: null,
          buffered: [],
          bufferedIndex: 0,
          allBuffers: true,
          allNoop: true,
          pendingcb: 0,
          constructed: true,
          prefinished: false,
          errorEmitted: false,
          emitClose: true,
          autoDestroy: true,
          errored: null,
          closed: false,
          closeEmitted: false,
          [Symbol(kOnFinished)]: []
        },
        _events: [Object: null prototype] {
          response: [Function: handleResponse],
          error: [Function: handleRequestError],
          socket: [Function: handleRequestSocket]
        },
        _eventsCount: 3,
        _maxListeners: undefined,
        _options: {
          maxRedirects: 21,
          maxBodyLength: 10485760,
          protocol: 'http:',
          path: '/api/protocol-fee-chart?timeframe=1D&endTimestamp=1688947198',
          method: 'GET',
          headers: [Object],
          agent: undefined,
          agents: [Object],
          auth: undefined,
          hostname: 'localhost',
          port: '3000',
          nativeProtocols: [Object],
          pathname: '/api/protocol-fee-chart',
          search: '?timeframe=1D&endTimestamp=1688947198'
        },
        _ended: true,
        _ending: true,
        _redirectCount: 0,
        _redirects: [],
        _requestBodyLength: 0,
        _requestBodyBuffers: [],
        _onNativeResponse: [Function (anonymous)],
        _currentRequest: ClientRequest {
          _events: [Object: null prototype],
          _eventsCount: 7,
          _maxListeners: undefined,
          outputData: [],
          outputSize: 0,
          writable: true,
          destroyed: false,
          _last: true,
          chunkedEncoding: false,
          shouldKeepAlive: false,
          maxRequestsOnConnectionReached: false,
          _defaultKeepAlive: true,
          useChunkedEncodingByDefault: false,
          sendDate: false,
          _removedConnection: false,
          _removedContLen: false,
          _removedTE: false,
          strictContentLength: false,
          _contentLength: 0,
          _hasBody: true,
          _trailer: '',
          finished: true,
          _headerSent: true,
          _closed: false,
          socket: [Socket],
          _header: 'GET /api/protocol-fee-chart?timeframe=1D&endTimestamp=1688947198 HTTP/1.1\r\n' +
            'Accept: application/json, text/plain, */*\r\n' +
            'User-Agent: axios/0.27.2\r\n' +
            'Host: localhost:3000\r\n' +
            'Connection: close\r\n' +
            '\r\n',
          _keepAliveTimeout: 0,
          _onPendingData: [Function: nop],
          agent: [Agent],
          socketPath: undefined,
          method: 'GET',
          maxHeaderSize: undefined,
          insecureHTTPParser: undefined,
          joinDuplicateHeaders: undefined,
          path: '/api/protocol-fee-chart?timeframe=1D&endTimestamp=1688947198',
          _ended: false,
          res: null,
          aborted: false,
          timeoutCb: null,
          upgradeOrConnect: false,
          parser: null,
          maxHeadersCount: null,
          reusedSocket: false,
          host: 'localhost',
          protocol: 'http:',
          _redirectable: [Circular *1],
          [Symbol(kCapture)]: false,
          [Symbol(kBytesWritten)]: 0,
          [Symbol(kNeedDrain)]: false,
          [Symbol(corked)]: 0,
          [Symbol(kOutHeaders)]: [Object: null prototype],
          [Symbol(errored)]: null,
          [Symbol(kUniqueHeaders)]: null
        },
        _currentUrl: 'http://localhost:3000/api/protocol-fee-chart?timeframe=1D&endTimestamp=1688947198',
        [Symbol(kCapture)]: false
      }
    }
    
    
    
    
    Done in 15.83s.

File: fees/move-dollar.ts (hunk 1 above)
Comment (LawsonGraham) [reply fees/move-dollar.ts]: have made modifications so backfill is now possible - endTimestamp is the date which is considered, and timeframe is how far behind that timestamp we set our range to be

File: dexs/thalaswap/index.ts [hunk 2]
```diff
@@ -3,17 +3,16 @@ import { SimpleAdapter } from "../../adapters/types";
 import { CHAIN } from "../../helpers/chains";
 import { getUniqStartOfTodayTimestamp } from "../../helpers/getUniSubgraphVolume";
 
-const thalaDappURL = 'https://app.thala.fi/';
+const thalaDappURL = 'http://localhost:3000';
```
Comment (dtmkeng) [inline dexs/thalaswap/index.ts]: `localhost` ?

**Review by dtmkeng — CHANGES_REQUESTED** [review]

File: dexs/thalaswap/index.ts (hunk 2 above)
Comment (LawsonGraham) [reply dexs/thalaswap/index.ts]: fixed!

**Comment by llamatester** [comment]
The thalaswap adapter exports: 
        
 
 _______________________________________
    Dexs for 9/7/2023
    _______________________________________
    
    APTOS 👇
    Backfill start time: 4/4/2023
    NO METHODOLOGY SPECIFIED
    Total volume: 49048701.84565541
    Daily volume: 130656.74801600662
    Total fees: 97693.59282484314
    Daily fees: 233.37112420354643
    Total protocol revenue: 19538.718560419416
    Daily protocol revenue: 46.674224829842096
    Timestamp: 1688947198
    
    
    
    
    Done in 5.96s.

**Comment by llamatester** [comment]
The move-dollar.ts adapter exports: 
        
 
 _______________________________________
    Fees for 9/7/2023
    _______________________________________
    
    APTOS 👇
    Backfill start time: 4/4/2023
    NO METHODOLOGY SPECIFIED
    Total fees: 42162.63796767994
    Daily fees: 1.2813749939203263
    Timestamp: 1688947198
    
    
    
    
    Done in 4.24s.

**Pull request merged**